src/pages/workspaces/workspace/jobHistory/CallCacheWizard.js

```diff
@@ -0,0 +1,284 @@
+    setDiffError(undefined)
+  }
+
+  const resetCallSelection = () => {
+    resetDiffResult()
+    setOtherCallFqnDropdownValue(undefined)
+    setOtherIndexDropdownValue(undefined)
+    setOtherCallSelected(false)
+  }
+
+  const resetWorkflowSelection = (value = undefined) => {
+    resetCallSelection()
+    setOtherWorkflowId(value)
+    setOtherWorkflowIdTextboxValue(value)
+    setMetadataFetchError(undefined)
+  }
+
+  const resetLink = resetAction => h(Link, { style: { fontSize: '14px', justifyContent: 'right' }, onClick: resetAction }, ['Reset'])
+
+  /*
+   * Page render
+   */
+
+  const divider = hr({ style: { width: '100%', marginTop: '2rem', marginBottom: '2rem', border: '1px ridge lightgray' } })
+
+  const step1 = () => {
+    return h(Fragment, [
+      div({ style: { paddingTop: '0.5rem', fontSize: 16, fontWeight: 500 } }, ['Step 1: Select the workflow you expected to cache from']),
+      div({ style: { marginTop: '0.5rem', marginBottom: '1rem', display: 'flex', flexDirection: 'row', alignItems: 'center' } }, [
+        h(IdContainer, [id => h(Fragment, [
+          label({ htmlFor: id, style: { paddingRight: '0.5rem' } }, ['Workflow ID:']),
+          div({ style: { paddingRight: '0.5rem', flex: '1' } }, [
+            h(TextInput, {
+              id,
+              style: Style.codeFont,
+              value: otherWorkflowIdTextboxValue,
+              onChange: setOtherWorkflowIdTextboxValue
+            })
+          ])
+        ])]),
+        h(ButtonPrimary, {
+          disabled: _.isEmpty(otherWorkflowIdTextboxValue),
+          onClick: () => {
+            resetWorkflowSelection(otherWorkflowIdTextboxValue)
+            readCalls(otherWorkflowIdTextboxValue)
+          }
+        }, ['Continue'])
+      ]),
+      metadataFetchError && [
+        'Error loading workflow metadata',
+        h(ErrorView, { error: metadataFetchError })
+      ]
+    ])
+  }
+
+  const step2 = () => {
+    const selectedCallIndex = Utils.cond(
+      [otherIndexDropdownValue !== undefined, () => otherIndexDropdownValue],
+      [otherWorkflowMetadata && otherCallFqnDropdownValue && otherWorkflowMetadata.calls[otherCallFqnDropdownValue].length === 1,
+        () => otherWorkflowMetadata.calls[otherCallFqnDropdownValue][0].shardIndex],
+      () => undefined
+    )
+
+    const otherCallSucceeded = otherWorkflowMetadata && otherCallFqnDropdownValue && selectedCallIndex !== undefined &&
+      _.some({ shardIndex: selectedCallIndex, executionStatus: 'Done' }, otherWorkflowMetadata.calls[otherCallFqnDropdownValue])
+
+    return h(Fragment, [
+      div({ style: { display: 'flex', alignItems: 'center', fontSize: 16, fontWeight: 500 } }, [
+        div(['Selected workflow B: ']),
+        div({ style: { display: 'flex', flexDirection: 'row', alignItems: 'center', flex: '1' } }, [
+          div({ style: { marginLeft: '0.5rem', ...Style.codeFont } }, otherWorkflowId)
+        ]),
+        resetLink(() => resetWorkflowSelection(undefined))
```

Comment:
We should reset to `''` rather than `undefined`, so the text is consistently a string.